### ISSUE TITLE
Custom icons on data elements in tree definitions not supporting publication statuses

### ISSUE THREAD

**Issue by wwwysocki** [Orckestra/CMS-Foundation#343]
**Summary**: When a custom icon is used on a publishable data element presented via a tree definition, it's impossible to tell a published item from an unpublished one.

**Repro**:

1. Download and install the [Orckestra.Tests.PublishableDataElements package](https://github.com/Orckestra/CMS-Foundation/files/713407/Orckestra.Tests.PublishableDataElements.zip).  
2. In the Content perspective, locate and expand the "Vegetables" node.
3. Select one of the child elements (e.g "Kale") and click "Unpublish" on the toolbar.

**Expected**: The icon of the data element ("Kale" in this case) will change its color to indicate its publication status: unpublished / draft.

**Actual**: The color of the icon remains unchanged (default).

**Notes**:

* The default icon - that of a data element - supports publication statuses by changing colors.
* There are publication statuses other than "published" and "unpublished"

![custom-icons-on-publishable-data-elements](https://cloud.githubusercontent.com/assets/14821915/22056975/69f7f2e8-dd6b-11e6-8388-51f91d1be909.png)

**Comment by burningice2866** [comment]
Yeah, this one always annoyed me. As i see it the issue is that there is no concept of different states of the same icon, so the Draft version of the page/data is literally a different icon, instead of applying some kind of filter to the main icon representing the node.
